apps/src/templates/ShareWarningsDialog.jsx

```diff
@@ -0,0 +1,79 @@
+var ShareWarnings = require('./ShareWarnings.jsx');
+
+/**
+ * Modal for our SharingWarnings.
+ */
+var SharingWarningsDialog = module.exports = React.createClass({
+  propTypes: {
+    is13Plus: React.PropTypes.bool.isRequired,
+    showStoreDataAlert: React.PropTypes.bool.isRequired,
+    handleClose: React.PropTypes.func.isRequired,
+    handleTooYoung: React.PropTypes.func.isRequired
+  },
+
+  getInitialState: function() {
+    return { modalIsOpen: !this.props.is13Plus || this.props.showStoreDataAlert };
+  },
+
+  componentDidMount: function () {
+    // We didn't need to show our modal. Go through the close process so that
+    // app becomes unblocked
+    if (!this.state.modalIsOpen) {
+      this.handleClose();
+    }
+  },
+
+  handleClose: function() {
+    this.setState({modalIsOpen: false});
+    React.unmountComponentAtNode(this.getDOMNode().parentNode);
+    this.props.handleClose();
+  },
+
+  render: function () {
+    if (!this.state.modalIsOpen) {
+      return <div/>;
```

Comment:
It might be that I don't need to call it at all. I'll poke into this. In both of the cases you list though, it should be called. In the case where modalIsOpen starts out false, componentDidMount will call handleClose for us.